.github/workflows/pr-main-build.yml

```diff
@@ -37,7 +37,32 @@ jobs:
       # TODO: cache single container build across matrix runs
```

Comment:
We probably need to look at an optimisation and decision on CICD, maybe another thing to discuss